### PR TITLE
fix: sidebar search crashes

### PR DESCRIPTION
**Pull request by lukaszcz**
## Summary
- prevent panics on Enter or arrow keys with empty results by clamping the cursor and checking selection.
- fix the crash path triggered by hiding the file tree, entering search, pressing Enter, and showing the tree again
- render the search sidebar correctly even when the file tree itself is hidden
- add regression tests for empty search results and hidden-sidebar search flows

### CODE REVIEW

File: pkg/ui/tui.go [hunk 1]
```diff
@@ -216,7 +217,7 @@ func (m mainModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 		tWidth, tHeight := m.sidebarWidth(), m.mainContentHeight()-searchHeight
 
 		m.fileTree.SetSize(tWidth, tHeight)
-		m.search.SetWidth(tWidth)
+		m.search.SetWidth(m.searchWidth())
```
[the remark this answers is not in the document]
Comment (lukaszcz) [reply pkg/ui/tui.go]: `m.searchWidth()` is not the same as `tWidth` (if I understood the comment correctly). This change is not necessary to fix the crashes, but makes the interface consistent: other search-width update paths always use `m.searchWidth()`. Feel free to remove the change and use `tWidth` instead. Makes little visual difference.

File: pkg/ui/tui.go [hunk 2]
```diff
@@ -775,4 +781,26 @@ func (m *mainModel) setSearchResults() {
...
 	m.filtered = filtered
+	switch {
+	case len(m.filtered) == 0:
+		m.resultsCursor = 0
+	case m.resultsCursor < 0:
+		m.resultsCursor = 0
+	case m.resultsCursor >= len(m.filtered):
+		m.resultsCursor = len(m.filtered) - 1
+	}
+}
```
[the remark this answers is not in the document]
Comment (lukaszcz) [reply pkg/ui/tui.go]: Not exactly correct, it's off by one when `m.resultsCursor >= len(m.filtered)` and `len(m.filtered) > 0`. One could fix the one-liner by changing `len(m.filtered)` to `max(0, len(m.filtered) - 1)`, but this gets complicated. Feel free to change to the fixed one-liner if you prefer it.

**Review by dlvhdr — APPROVED** [review]
thank you!
